From d0fe6e20c86b42be642caee1a6aa0c786bdb8ecb Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Mon, 1 Feb 2016 16:15:36 -0500
Subject: [PATCH 1320/1565] drm/amd/dal: Clean up MST stream on our ASIC

Make sure we call HWSS to clean up our ASIC's MST
registers. deallocate_mst_payload will guard calls
into drm_mst.

Change-Id: I6a1bcd4fbe57ea91f152197aaa85f772c3b624ed
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index b180cf6..2928724 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -1634,8 +1634,7 @@ void core_link_disable_stream(
 {
 	struct dc *dc = stream->ctx->dc;
 
-	if (stream->signal == SIGNAL_TYPE_DISPLAY_PORT_MST &&
-		link->public.type == dc_connection_mst_branch)
+	if (stream->signal == SIGNAL_TYPE_DISPLAY_PORT_MST)
 		deallocate_mst_payload(stream);
 
 	dc->hwss.disable_stream(stream);
-- 
1.9.1

